tensorboard/webapp/runs/views/runs_table/regex_edit_dialog_test.ts

```diff
@@ -254,4 +264,74 @@ describe('regex_edit_dialog', () => {
     const input = fixture.debugElement.query(By.css('input'));
     expect(input.nativeElement.value).toBe('(train|eval)');
   });
+
+  describe('live grouping result preview', () => {
+    it('does not render grouping result initially', () => {
+      const fixture = createComponent(['rose']);
+      fixture.detectChanges();
+
+      const groupingResult = fixture.debugElement.query(
+        By.css('.group-container')
+      );
+      expect(groupingResult).toBeNull();
+    });
+
+    it('renders grouping preview on regex query input', fakeAsync(() => {
```

Comment:
Would you mind adding a small test that tests what we do with an invalid regex string? Examples of "invalid" regex string are:
- empty string
- invalid regex
- no match

Reply:
I'm not sure about the second item "invalid regex" do you mean if the input is just a string but not regex string?
for example `train` vs `train_(\d+)`
the first case I think it's okay to still render a group (and only one group)
if not, please provide some examples to make my test cases clear. Thanks!

Reply:
`train_(\d+` would be a broken regex. `\` would be another one.